sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1518,6 +1518,166 @@ case class ArrayContains(left: Expression, right: Expression)
     copy(left = newLeft, right = newRight)
 }
 
+/**
+ * Searches the specified array for the specified object using the binary search algorithm.
+ * This expression is dedicated only for PySpark and Spark-ML.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
```

Comment:
Is `@ExpressionDescription` really necessary for Internal Expressions? Since they do not exist in `sql-expression-schema.md`, it seems there's no necessity for it. 

Moreover, the currently registered Internal Expressions are not consistent in this regard. Most of them do not have the `@ExpressionDescription` annotation, but a few are annotated, such as `TimestampAdd` and `TimestampDiff`. Should we unify the format? 

Additionally, should Internal Expressions be placed under a separate package?